src/quote/flashmint/hyeth/provider.ts

```diff
@@ -0,0 +1,81 @@
+import { Contract } from '@ethersproject/contracts'
+
+import FLASHMINT_HYETH_ABI from 'constants/abis/FlashMintHyEth.json'
+import { FlashMintHyEthAddress } from 'constants/contracts'
+import { QuoteProvider, QuoteToken } from 'quote/interfaces'
+import { getRpcProvider } from 'utils/rpc-provider'
+import { SwapData, wei } from 'utils'
+
+import {
+  getComponentsSwapData,
+  getEthToInputOutputTokenSwapData,
+  getInputTokenToEthSwapData,
+} from './swap-data'
+import { BigNumber } from '@ethersproject/bignumber'
+
+export interface FlashMintHyEthQuoteRequest {
+  isMinting: boolean
+  inputToken: QuoteToken
+  outputToken: QuoteToken
+  indexTokenAmount: bigint
+  slippage: number
+}
+
+export interface FlashMintHyEthQuote {
+  indexTokenAmount: bigint
+  inputOutputTokenAmount: bigint
+  // Represents `swapDataEthToComponent` for minting
+  // and `swapDataComponentToEth` for redeeming
+  componentsSwapData: SwapData[]
+  // Used only for minting w/ ERC-20 tokens
+  swapDataInputTokenToEth: SwapData | null
+  // Represents `swapDataEthToInputToken` for minting w/ ERC-20 token
+  // and `swapDataEthToOutputToken` for redeeming to ERC-20 token
+  swapDataEthToInputOutputToken: SwapData | null
+}
+
+export class FlashMintHyEthQuoteProvider
+  implements QuoteProvider<FlashMintHyEthQuoteRequest, FlashMintHyEthQuote>
+{
+  constructor(private readonly rpcUrl: string) {}
+
+  async getQuote(
+    request: FlashMintHyEthQuoteRequest
+  ): Promise<FlashMintHyEthQuote | null> {
+    const provider = getRpcProvider(this.rpcUrl)
+    const { indexTokenAmount, inputToken, isMinting, outputToken } = request
+    const componentsSwapData = getComponentsSwapData(isMinting)
+    // Only relevant for minting ERC-20's
+    const swapDataInputTokenToEth = isMinting
+      ? getInputTokenToEthSwapData(inputToken)
+      : null
+    const inputOutputToken = isMinting ? inputToken : outputToken
+    const swapDataEthToInputOutputToken =
+      getEthToInputOutputTokenSwapData(inputOutputToken)
+    // TODO: static call write functions?
+    const indexToken = isMinting ? outputToken : inputToken
+    const contract = new Contract(
+      FlashMintHyEthAddress,
+      FLASHMINT_HYETH_ABI,
+      provider
+    )
+    // TODO: switch to provider.call(tx) from builder to handle issue/redeem
+    // TODO: just for testing, delete later
+    const inputOutputTokenAmount: BigNumber =
+      await contract.callStatic.issueExactSetFromETH(
+        indexToken.address,
+        indexTokenAmount,
+        componentsSwapData,
+        // TODO:
+        { value: wei(1) }
+      )
+    console.log(inputOutputTokenAmount.toString())
```

Comment:
This is just to test one case for now (issueExactSetFromETH). The question here is... usually, we calculate the input amount from the swap data quotes - which we don't have in this case.

* for `minting` we could add the inputTokenAmount to the request object (FlashMintHyEthQuoteRequest)
* for `redeeming` we'd still have the challenge of not having a way to determine the output amount

Is there a way to just put an arbitrary value for maxAmountIn or minAmountOut? @ckoopmann

Reply:
As the token is not rebalance yet, this is failing with: ` call revert exception; VM Exception while processing transaction: reverted with reason string "Missing Swapdata for non-standard component" [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="issueExactSetFromETH(address,uint256,(address[],uint24[],address,uint8)[])", data="0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002b4d697373696e6720537761706461746120666f72206e6f6e2d7374616e6461726420636f6d706f6e656e74000000000000000000000000000000000000000000", errorArgs=["Missing Swapdata for non-standard component"], errorName="Error", errorSignature="Error(string)", reason="Missing Swapdata for non-standard component", code=CALL_EXCEPTION, version=abi/5.7.0)`

Or does this indicate another problem? @ckoopmann